Add KPIs for system's autonomy to scenarios_and_results.rst

MaGering opened the pull request:
With this PR a description of the KPIs, which are used to describe the system's autonomy and to discuss the results in pvcompare's simulation of scenarios, is added to the documentation.

The following steps were realized, as well (if applies):
:x: Use in-line comments to explain your code
:x: Write docstrings to your code
:x: Write test(s) for your new patch of code
- [x] For new functionalities: Explain in readthedocs
- [x] Update the CHANGELOG.md (let's start this after the first release)
- [x] Apply black (`black . --exclude docs/`)
- [x] Check if full simulation tests pass locally (`EXECUTE_TESTS_ON=master pytest`)


*Please mark above checkboxes as following:*
- [ ] Open
- [x] Done

:x: Check not applicable to this PR

<sub>*For more information on how to contribute check the [CONTRIBUTING.md](https://github.com/rl-institut/mvs_eland/blob/dev/CONTRIBUTING.md).*<sub>

MaGering commented:
@SabineHaas can you help me with the definition of the `self consumption`? 

![grafik](https://user-images.githubusercontent.com/52790556/112001918-ac8b7600-8b1f-11eb-9c80-b930eaa1cf1a.png)

The one I added to the RTD with this PR (see equation above), is the one I found in the RTD of MVS (except for `E_excess` missing, but I think it must be considered doesn't it?):

![grafik](https://user-images.githubusercontent.com/52790556/112000639-78638580-8b1e-11eb-8dd8-9e7d1fd65ccc.png)

And this is what's implemented in out post-processing right now:

```
file_sheet3.at["Onsite energy fraction", 0] = (
            file_sheet3.at["Total_demandElectricity", 0]
            - file_sheet3.at["Total_feedinElectricity", 0]
        ) / file_sheet3.at["Total internal renewable generation", 0]
```
Unfortunately I can no longer understand this definition, because the only other definition I would understand (defining over the demand) would be this one:

![grafik](https://user-images.githubusercontent.com/52790556/112001376-3424b500-8b1f-11eb-822f-92b57cc17a0a.png)

SabineHaas commented:
> The one I added to the RTD with this PR (see equation above), is the one I found in the RTD of MVS (except for `E_excess` missing, but I think it must be considered doesn't it?):

I agree, it should be considered.



> And this is what's implemented in out post-processing right now:
> 
> ```
> file_sheet3.at["Onsite energy fraction", 0] = (
>             file_sheet3.at["Total_demandElectricity", 0]
>             - file_sheet3.at["Total_feedinElectricity", 0]
>         ) / file_sheet3.at["Total internal renewable generation", 0]
> ```
> 
> Unfortunately I can no longer understand this definition, because the only other definition I would understand (defining over the demand) would be this one:
> 
> ![grafik](https://user-images.githubusercontent.com/52790556/112001376-3424b500-8b1f-11eb-822f-92b57cc17a0a.png)

I also agree here.. as self-consumption is the share of local generation used (internally) to cover the demand. --> the equation has to be `generation_used_in_the_system/generation`.

generation_used_in_the_system is `demand - consumption` not `demand - feed-in`.

MaGering commented:
> I also agree here.. as self-consumption is the share of local generation used (internally) to cover the demand. --> the equation has to be `generation_used_in_the_system/generation`.
> 
> generation_used_in_the_system is `demand - consumption` not `demand - feed-in`.

Nice. I would however implement the self consumption using the first definition in the post-processing and RTD: 

![grafik](https://user-images.githubusercontent.com/52790556/112001918-ac8b7600-8b1f-11eb-9c80-b930eaa1cf1a.png)

because it gives us a formulation that is independent of the demand and thus of the energy carrier. This leads to a comparability of sector coupled scenarios. Do you agree @SabineHaas?

SabineHaas commented:
> > I also agree here.. as self-consumption is the share of local generation used (internally) to cover the demand. --> the equation has to be `generation_used_in_the_system/generation`.
> > generation_used_in_the_system is `demand - consumption` not `demand - feed-in`.
> 
> Nice. I would however implement the self consumption using the first definition in the post-processing and RTD:
> 
> ![grafik](https://user-images.githubusercontent.com/52790556/112001918-ac8b7600-8b1f-11eb-9c80-b930eaa1cf1a.png)
> 
> because it gives us a formulation that is independent of the demand and thus of the energy carrier. This leads to a comparability of sector coupled scenarios. Do you agree @SabineHaas?

Yes, I agree. You're right in case of self-consumption there's no need to use the demand for which we need a workaround (total demand = elec demand + elec demand of heat pump), as we can just use generation in the expression :+1:

MaGering commented:
This PR is ready for review @SabineHaas.

SabineHaas commented on docs/scenarios_and_results.rst:
```diff
@@ -6,6 +6,38 @@ GRECO Results
+Definition of KPIs
+------------------
+
+`Self-consumption`_  also *onsite energy fraction* is defined as the fraction of all locally generated energy that is consumed by the system itself to the system's total generation:
```
```suggestion
`Self-consumption`_  also *onsite energy fraction* is defined as the fraction of all locally generated energy that is consumed by the system itself to the system's total local generation:
```

SabineHaas commented on docs/scenarios_and_results.rst:
```diff
@@ -6,6 +6,38 @@ GRECO Results
+.. _Self-consumption:
+
+.. math::
+        Self Consumption &=\frac{\sum_{i} {E_{generation} (i)} - E_{gridfeedin}(i) - E_{excess}(i)}{\sum_{i} {E_{generation} (i)} }
```
```suggestion
        Self\_Consumption &=\frac{\sum_{i} {E_{generation} (i)} - E_{gridfeedin}(i) - E_{excess}(i)}{\sum_{i} {E_{generation} (i)} }
```
I would write all words (also below) like this, otherwise the letters look quite strange in RTD.

SabineHaas reviewed (commented):
Thank you @MaGering for fixing and for the description :+1: 

I wonder whether the description fits better to "Simulation outputs".. you can decide that. I guess the section "simulations outputs" will be filled further, stating which files and which information you can find in the outputs - that's why i think the KPIs could fit there.

MaGering commented on docs/scenarios_and_results.rst:
```diff
@@ -6,6 +6,38 @@ GRECO Results
+       Degree Of Net Zero Energy = 1 + \frac{\sum_{i} E_{gridfeedin}(i) - E_{consumption,provider}(i)}{\sum_{i} E_{demand} (i)}
+
+
+Please see the section `Outputs of the MVS simulation / Technical data <https://multi-vector-simulator.readthedocs.io/en/latest/MVS_Outputs.html#technical-data>`_ of MVS's documentation to learn more about the `degree of net zero energy`_.
```
Yes! Thank you. :) Done with commit https://github.com/greco-project/pvcompare/pull/261/commits/1bbb2ce45980b68d28d8669fb5dc6368efcd7c88.

MaGering commented:
> Thank you @MaGering for fixing and for the description +1
> 
> I wonder whether the description fits better to "Simulation outputs".. you can decide that. I guess the section "simulations outputs" will be filled further, stating which files and which information you can find in the outputs - that's why i think the KPIs could fit there.

Sounds reasonable! Changed in commit https://github.com/greco-project/pvcompare/pull/261/commits/1bbb2ce45980b68d28d8669fb5dc6368efcd7c88. I also added a sentence as introduction of this section.

SabineHaas reviewed (approved):
> Sounds reasonable! Changed in commit [1bbb2ce](https://github.com/greco-project/pvcompare/commit/1bbb2ce45980b68d28d8669fb5dc6368efcd7c88). I also added a sentence as introduction of this section.

:+1: 

I don't see the introduction sentence though - maybe you forgot to commit?

Before merging: Could you add the changes to the changelog.md please and confirm with the checkbox above that you have run `EXECUTE_TESTS_ON=master pytest` without errors?

Thank you :)

MaGering commented:
> I don't see the introduction sentence though - maybe you forgot to commit?

It's the first sentence in section `Definition of KPIs` that is new. :)

SabineHaas commented:
> > I don't see the introduction sentence though - maybe you forgot to commit?
> 
> It's the first sentence in section `Definition of KPIs` that is new. :)

:see_no_evil: sorry ;)
